tests/integration/models/test_spatio_temporal_variational.py

```diff
@@ -0,0 +1,125 @@
+#
+# Copyright (c) 2021 The Markovflow Contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Module containing the integration tests for children of the `SpatioTemporalBase` class."""
+from copy import deepcopy
+import pytest
+import numpy as np
+import tensorflow as tf
+from gpflow.likelihoods import Gaussian
+import gpflow.kernels
+from gpflow.models import GPR
+
+from markovflow.kernels import Matern12
+from markovflow.models import (
+    SpatioTemporalSparseVariational,
+    SpatioTemporalSparseCVI,
+)
+from markovflow.models.spatio_temporal_variational import SpatioTemporalBase
+
+
+@pytest.fixture(name="st_data_rng")
+def _setup_data_rng_fixture():
+    rng = np.random.default_rng(42)
+    return rng
+
+
+@pytest.fixture(name="st_data")
+def _setup_spatiotemporal_data_fixture(st_data_rng):
+    x = np.array([0.0, 1.0])
+    t = np.array([2.0, 3.0])
+    X = np.array(np.meshgrid(x, t)).T.reshape(-1, 2)
+    X = X[np.argsort(X[:, 1]), :]
+    Y = st_data_rng.normal(size=(X.shape[0], 1))
+    return (X, Y)
+
+
+@pytest.fixture(name="st_model_params")
+def _setup_spatiotemporal_params_fixture(st_data):
+    X, _ = st_data
+    params = dict(
+        inducing_space=np.unique(X[:, 0]).reshape(-1, 1),
+        inducing_time=np.unique(X[:, 1]),
+        kernel_space=gpflow.kernels.Matern32(lengthscales=1, variance=1, active_dims=[0]),
+        kernel_time=Matern12(lengthscale=1, variance=1),
+        likelihood=Gaussian(),
```

Comment:
maybe you want to set the learning_rate=1
currently it is set to 0.1 and I m not sure it converges in the 10 steps you have here.

Reply:
The learning rate is set to 1 here: https://github.com/secondmind-labs/markovflow/blob/5229ac63e8e18856c59002a73404161f2e536c49/tests/integration/models/test_spatio_temporal_variational.py#L111

It seems to converge for ELBO, mean, and variance after a single step. Running even for 1,000 steps gives no different result than 1 step.